cypress/integration/tests/signUp.js

```diff
@@ -0,0 +1,95 @@
+
+describe('Sign up tests', () => {
+  beforeEach(() => {
+    SigningPage.open('/signing');
+  });
+
+  it('TA-8: User is able to sign up using valid data', () => {
+    const emailAddress = generateSignUpEmail();
+    SigningPage.signUp(name.valid, emailAddress, password.valid, password.valid);
+    SigningPage.signOutIcon.should('be.visible');
+  });
+
+  it('TA-8.1: User unable to sign up with already registered email', () => {
+    SigningPage.signUp(name.valid, email.registered, password.valid, password.valid);
+    SigningPage.signUpError.should('contain', errorMessages.alreadyRegisteredEmail);
+  });
+
+  it('TA-9: User unable to sign up with missing Name', () => {
+    SigningPage.signUp(undefined, email.registered, password.valid, password.valid);
+    SigningPage.emptySignUpNameField.should('have.length', 1);
+    SigningPage.emptySignUpNameField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-10: User unable to sign up with missing Email', () => {
+    SigningPage.signUp(name.valid, undefined, password.valid, password.valid);
+    SigningPage.emptyOrInvalidSignUpEmailField.should('have.length', 1);
+    SigningPage.emptyOrInvalidSignUpEmailField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-11: User unable to sigh up with missing Password', () => {
+    SigningPage.signUp(name.valid, email.registered, undefined, password.valid);
+    SigningPage.emptyPasswordField.should('have.length', 1);
+    SigningPage.emptyPasswordField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-12: User unable to sign up with missing Confirm Password', () => {
+    SigningPage.signUp(name.valid, email.registered, password.valid, undefined);
+    SigningPage.emptyConfirmPasswordField.should('have.length', 1);
+    SigningPage.emptyConfirmPasswordField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.emptyEmailOrPasswordOrNameField);
+    });
+  });
+
+  it('TA-13: User unable to sign up if Confirm Password is not matching Password', () => {
+    SigningPage.signUp(
+      name.valid,
+      email.unregistered,
+      password.valid,
+      confirmPassword.unequalToPassword,
+    );
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.eq(errorMessages.passwordDoesntMatch);
+    });
+  });
+
+  it('TA-15: User unable to sign up with invalid Email', () => {
+    SigningPage.signUp(name.valid, email.invalid, password.valid, password.valid);
+    SigningPage.emptyOrInvalidSignUpEmailField.should('have.length', 1);
+    SigningPage.emptyOrInvalidSignUpEmailField.then(($input) => {
+      expect($input[0].validationMessage).to.eq(errorMessages.invalidEmailDoubledAtSign);
+    });
```

Comment:
I don't really the idea of adding `:invalid` to locators and creating `emptyOrInvalid...` objects.
What is on my mind: you don't need to exactly match the place of where this error happens but just the fact that the error message appears.
So please rework: assert that the field validation appeared in general on the form and particular error message is up.